src/parser.cpp

```diff
@@ -319,10 +380,21 @@ namespace ada::parser {
           // - url’s port to base’s port,
           // - state to path state, and then, decrease pointer by 1.
           else {
-            url.username = base_url->username;
-            url.password = base_url->password;
-            url.host = base_url->host;
-            url.port = base_url->port;
+            if constexpr (result_type_is_ada_url) {
+              url.username = base_url->username;
+              url.password = base_url->password;
+              url.host = base_url->host;
+              url.port = base_url->port;
+            } else {
+              // TODO: This is likely not correct. There is a difference between having an empty username
+              // and not having a username.
+              url.update_base_username(base_url->get_username());
```

Comment:
Same issue here, we call update_base_username even if there is no username in the base.